src/Service.Tests/Configuration/TelemetryTests.cs

```diff
@@ -0,0 +1,295 @@
+
+        string[] args = new[]
+        {
+            $"--ConfigFileName={CONFIG_WITH_TELEMETRY}"
+        };
+
+        List<ITelemetry> telemetryItems = new();
+        ITelemetryChannel telemetryChannel = new CustomTelemetryChannel(telemetryItems);
+        Startup.CustomTelemetryChannel = telemetryChannel;
+        TestServer server = new(Program.CreateWebHostBuilder(args));
+
+        await TestRestAndGraphQLRequestsOnServerInNonHostedScenario(server);
+
+        server.Dispose();
+        ((CustomTelemetryChannel)Startup.CustomTelemetryChannel).Flush();
+
+        Console.WriteLine(JsonSerializer.Serialize(telemetryItems));
+
+        // Assert that we are sending Traces/Requests/Exceptions
+        Assert.IsTrue(telemetryItems.Any(item => item is TraceTelemetry));
+        Assert.IsTrue(telemetryItems.Any(item => item is RequestTelemetry));
+        Assert.IsTrue(telemetryItems.Any(item => item is ExceptionTelemetry));
+
+        // Asserting on count Exception/Request telemetry items.
+        Assert.AreEqual(1, telemetryItems.Count(item => item is ExceptionTelemetry));
+        Assert.AreEqual(2, telemetryItems.Count(item => item is RequestTelemetry));
+
+        Assert.IsTrue(telemetryItems.Any(item =>
+            item is RequestTelemetry
+            && ((RequestTelemetry)item).Name.Equals("POST /graphql")
+            && ((RequestTelemetry)item).ResponseCode.Equals("200")
+            && ((RequestTelemetry)item).Url.PathAndQuery.Equals("/graphql")));
+
+        Assert.IsTrue(telemetryItems.Any(item =>
+        item is RequestTelemetry
+        && ((RequestTelemetry)item).Name.Equals("POST Rest/Insert [route]")
+            && ((RequestTelemetry)item).ResponseCode.Equals("403")
+            && ((RequestTelemetry)item).Url.PathAndQuery.Equals("/api/Publisher/id/1?name=Test")));
+
+        Assert.IsTrue(telemetryItems.Any(item =>
+        item is ExceptionTelemetry
+        && ((ExceptionTelemetry)item).Message.Equals("Authorization Failure: Access Not Allowed.")));
+    }
+
+    /// <summary>
+    /// Testing the Hosted Scenario for both configuration endpoint.
+    /// Tests that different telemetry items such as Traces or logs, Exceptions and Requests
+    /// are correctly sent to application Insights when enabled.
+    /// Also asserting on their respective properties.
+    /// </summary>
+    [DataTestMethod]
+    [DataRow(CONFIGURATION_ENDPOINT)]
+    [DataRow(CONFIGURATION_ENDPOINT_V2)]
+    public async Task TestTelemetryItemsAreSentCorrectly_HostedScenario(string configurationEndpoint)
+    {
+        // Disable parallel execution for this test method
+        TestContext.Properties.Add("ParallelScope", "Individual");
+
+        SetUpTelemetryInconfig(CONFIG_WITH_TELEMETRY, isTelemetryEnabled: true, TEST_APP_INSIGHTS_CONN_STRING);
+
+        string[] args = new[]
+        {
+            $"--ConfigFileName={CONFIG_WITH_TELEMETRY}"
+        };
```

Comment:
The `args` array doesn't seem to be needed and doesn't look to be used in this test.